Sources/GRPCHTTP2TransportNIOPosix/HTTP2ServerTransport+Posix.swift

```diff
@@ -315,12 +382,14 @@ extension HTTP2ServerTransport.Posix {
       compression: HTTP2ServerTransport.Config.Compression,
       connection: HTTP2ServerTransport.Config.Connection,
       http2: HTTP2ServerTransport.Config.HTTP2,
-      rpc: HTTP2ServerTransport.Config.RPC
+      rpc: HTTP2ServerTransport.Config.RPC,
+      tls: HTTP2ServerTransport.Config.TLS? = nil
```

Comment:
We definitely shouldn't default this to `nil`. If anything we should make it really clear when TLS isn't being used; maybe by having a wrapper type with two cases: `tis` and `plaintext`, then the option can be `transportSecurity: .plaintext` or `transportSecurity: .tls(myTLSConfig)`